server/server_test.go

```diff
@@ -586,3 +582,75 @@ func TestLoadSubmits(t *testing.T) {
 	cancel()
 	egSrv.Wait()
 }
+
+// Test submitting a challenge followed by proof generation and getting the proof via GRPC
+// in registration-only mode.
+// It should return an empty proof but with members populated.
+func TestRegistrationOnlyMode(t *testing.T) {
+	t.Parallel()
+	req := require.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logging.NewContext(ctx, zaptest.NewLogger(t))
+
+	cfg := server.DefaultConfig()
+	cfg.DisableWorker = true
+	cfg.PoetDir = t.TempDir()
+	cfg.Genesis = server.Genesis(time.Now())
+	cfg.Round.EpochDuration = time.Millisecond * 10
+	cfg.Round.PhaseShift = 0
+	cfg.Round.CycleGap = 0
+	cfg.RawRPCListener = randomHost
+	cfg.RawRESTListener = randomHost
+
+	srv, client := spawnPoet(ctx, t, *cfg)
+	t.Cleanup(func() { assert.NoError(t, srv.Close()) })
+
+	var eg errgroup.Group
+	eg.Go(func() error {
+		return srv.Start(ctx)
+	})
+
+	// Submit a challenge
+	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
+	req.NoError(err)
+	challenge := []byte("poet challenge")
+
+	powParams, err := client.PowParams(context.Background(), &api.PowParamsRequest{})
+	req.NoError(err)
+
+	signature := ed25519.Sign(privKey, challenge)
+	resp, err := client.Submit(context.Background(), &api.SubmitRequest{
+		Challenge: challenge,
+		Pubkey:    pubKey,
+		PowParams: powParams.PowParams,
+		Signature: signature,
+	})
+	req.NoError(err)
+
+	// Query for the proof
+	var proof *api.ProofResponse
+	req.Eventually(func() bool {
+		proof, err = client.Proof(context.Background(), &api.ProofRequest{RoundId: resp.RoundId})
+		return err == nil
+	}, time.Second, time.Millisecond*10)
+
+	req.Zero(proof.Proof.Leaves)
+	req.Len(proof.Proof.Members, 1)
+	req.Contains(proof.Proof.Members, challenge)
+	cancel()
+
+	merkleProof := shared.MerkleProof{
+		Root:         proof.Proof.Proof.Root,
+		ProvenLeaves: proof.Proof.Proof.ProvenLeaves,
+		ProofNodes:   proof.Proof.Proof.ProofNodes,
+	}
```

Comment:
NIT: `proof.Proof.Proof.X` seems quite redundant 😅

Reply:
yes... could be `response.Proof.Proof.X` :shrug: